transformer/mappings_helpers.py

```diff
@@ -14,3 +14,17 @@ def closest_parent_value(self, uri, key):
             ancestor = self.aspace.client.get(a['ref']).json()
             if ancestor.get(key) not in ['', [], {}, None]:
                 return ancestor[key]
+
+    def has_children(self, uri):
+        """Checks whether an archival object has children using the tree/node endpoint.
+        Checks the child_count attribute and if the value is greater than 0, return true, otherwise return False."""
+        obj = self.aspace.client.get(uri).json()
```

Comment:
So, I updated the code to use the abstraction layer. I think it works significantly slower than the older code though. By a few minutes, at least. It may be because I have to split the URI to get the AO ID, but you might have a better way of doing that. I just couldn't figure a different way out.

Reply:
Thanks for highlighting the performance issue - that's something I hadn't considered and is actually pretty important.  I don't know that simply splitting the URI would cause that performance hit. It is certainly possible that the abstraction layer is slower than just getting a URI because it's making multiple requests. I do think that's kind of a problem though, so thanks for pointing it out. A couple of minutes is going to increase exponentially once we do this at scale. Do you think we should go back to fetching just the URI?

Reply:
To add some hard data to the the equation, I ran both versions with the only change being whether we fetch the URI or use the abstraction layer.

Abstraction layer runtime: 3 minutes 44 seconds
Fetch URI runtime: 50 seconds

That's a pretty substantial difference, so I'd lean towards going back to fetching the URI.

Reply:
Yeah, I would agree with that - I think speed trumps less lines of code here.